lib/status-bar-manager.coffee

```diff
@@ -14,9 +19,10 @@ class StatusBarManager
 
   initialize: (@statusBar) ->
 
-  update: (currentMode) ->
+  update: (currentMode, currentSubmode) ->
+    currentFullMode = currentMode + "$" + currentSubmode if currentSubmode?
```

Comment:
I like it; nice solution. Nitpicks:

* I'd rather use `.` as the delimiter than `$`. It would just mean we'd have to quote the keys.
* I think you could avoid depending on the order of `ContentsByMode` if you did this:

```coffee
currentMode = currentMode + "." + currentSubmode if currentSubmode?
for mode, [klass, html] of ContentsByMode
  if mode is currentMode
    # ...  
```
```